app/code/Magento/OpenSearch/Model/SearchClient.php

```diff
@@ -142,6 +142,11 @@ private function buildOSConfig(array $options = []): array
         // @codingStandardsIgnoreEnd
         if (!$protocol) {
             $protocol = 'http';
+
+            // If port is 443 then use https
+            if (!empty($options['port']) && $options['port'] == '443') {
```

Comment:
Suggest you to implement strict check here like below:

`$options['port'] === '443'`